### ISSUE TITLE
训练模型剪枝时出错，出现数组越界错误

### ISSUE THREAD

**Issue by yezechen** [PaddlePaddle/PaddleSlim#899]
debug发现是在
    def _visit_and_search(self, var, axis, transforms):
        self._visit(var, axis)
        pre_ops = var.inputs()
        for op in pre_ops: 
这段代码循环到一个op，内容是{Out=['learning_rate']} = fill_constant(inputs={ShapeTensor=[], ShapeTensorList=[], ValueTensor=[]}, dtype = 5, force_cpu = False, op_device = , op_namescope = /, op_role = 16, op_role_var = [], place_type = -1, shape = [1], str_value = 0.001, value = 0.0010000000474974513)，这时候剪枝报错。
